specs/waku/v2/waku-swap-accounting.md

```diff
@@ -20,17 +21,54 @@ authors: Oskar Thorén <oskar@status.im>
 
 **Protocol identifier***: `/vac/waku/swap/2.0.0-alpha1`
 
 # Motivation
 
-The Waku network makes up a service network, and some nodes provide a useful service to other nodes. We want to account for that, laying the foundation for settlement.
+The Waku network makes up a service network, and some nodes provide a useful service to other nodes. We want to account for that, and when imbalances arise, settle this. The core of this approach has some theoretical backing in game theory, and variants of it have practically been proven to work in systems such as Bittorrent. The specific model use was developed by the Swarm project (previously part of Ethereum), and we re-use contracts that were written for this purpose.
 
-TODO: Fill in more
+By using a delayed payment mechanism in the form of cheques, a barter-like mechanism can arise, and nodes can decide on their own policy as opposed to be strictly tied to a specific payment scheme. Additionally, this delayed settlement eases requirements on the underlying network in terms of transaction speed or costs.
 
-TODO: Game theoretical foundation, Bittorrent, relaxed tit-for-tat
+Theoretically, nodes providing and using resources over a long, indefinite, period of time can be seen as a an iterated form of prisoner's dilemma. Specifically, and more intuitively, since we have a cost and benefit profile for each provision/usage (of Waku Message's, e.g.), and the pricing can be set such that mutual cooperation is incentivzed, this can be analyzed as a form of donations game.
+
+# Game Theory - Iterated prisoner's dilemma / donation game
+
+What follows is a sketch of what the game looks like between two nodes. We can
+look at it as a game special case of iterated prisoner's dilemma where:
+
+Each node can cooperate with some benefit at a personal cost c, where b>c. From A's point of view:
+
+A/B | Cooperate | Defect
+-----|----------|-------
+Cooperate | b-c | -c
+Defect | b | 0
+
+with the generalized form of PD being:
+
+A/B | Cooperate | Defect
+-----|----------|-------
+Cooperate | R | S
+Defect | T | P
+
+R=reward, S=Sucker's payoff, T=temptation, P=punishment
+
+And the following holds:
+
+- `T>R>P>S`
+- `2R>T+S`
+
+In our case, this means `b>b-c>0>-c` and `2(b-c)> b-c` which is trivially true.
+
+As this is an iterated game with no clear finishing point in most circumstances, a tit-for-tat strategy is simple, elegant and functional. To be more theoretically precise, this also requires reasonable assumptions on error rate and discount parameter. See Axelrod for more details. In specific circumstances, nodes can choose slightly different policies if there's a strong need for it.
```

Comment:
> In specific circumstances, nodes can choose slightly different policies if there's a strong need for it

It seems like nodes decide on a `policy`, it would be great to mention and define this somewhere earlier in the spec

Reply:
for now I added a quick follow up sentence, we can elaborate on it more later on when it becomes more relevant I think